src/Storages/VirtualColumnUtils.h

```diff
@@ -32,7 +32,7 @@ void buildSetsForDAG(const ActionsDAG & dag, const ContextPtr & context);
 bool isDeterministicInScopeOfQuery(const ActionsDAG::Node * node);
 
 /// Extract a part of predicate that can be evaluated using only columns from input_names.
-ActionsDAGPtr splitFilterDagForAllowedInputs(const ActionsDAG::Node * predicate, const Block * allowed_inputs);
+ActionsDAGPtr splitFilterDagForAllowedInputs(const ActionsDAG::Node * predicate, const Block * allowed_inputs, bool strict = false);
```

Comment:
Let's use better name for the flag, because `strict` doesn't tell anything. Something like `allow_non_deterministic_functions = true`. And let's add a comment here what this flag means and why we need it.